scalalib/src/mill/javalib/android/AndroidAppModule.scala

```diff
@@ -1030,11 +1076,15 @@ trait AndroidAppModule extends JavaModule {
 
     override def androidEmulatorPort: String = parent.androidEmulatorPort
 
-    override def sources: T[Seq[PathRef]] = parent.sources() :+ PathRef(androidTestPath / "java")
+    override def sources: T[Seq[PathRef]] = Seq(PathRef(androidTestPath / "java"))
```

Comment:
This is a good change, but still, the way instrumented testing is implemented right now is different from the way AGP does it.

The current approach is to generate a single fat APK, consisting of production + test files.

AGP is doing it in a different way: it will be 2 APKs, one is a normal production APK, another one will contain _only_ test files and will also have a dedicated manifest obviously, with a dedicated package name attribute which will be `${main app package name}.test`. And this test APK should be installed with `-t` option (see [here](https://developer.android.com/tools/adb#-t-option)).

Reply:
yes, you are right, I have decompiled the APKs from AGP as well and can work towards this if we want to have the exact same behaviour (which makes sense to have)

![image](https://github.com/user-attachments/assets/9a3ff5ca-5b9c-418b-abc5-449000ef140f)

I want to set up a scaffold with some tests and examples working first and then do the underlying implementations on the next stage.